### PR TITLE
[android] Fixed PeripheralBusAndroid crashing

### PR DESCRIPTION
**Pull request by dan1j3l**
Fixed crashing on android devices with API Level 19 and higher.
Bumped minimum required android SDK to version 19. (due to usage of functions from API 19).
Modified functions from API 21 to API 19.

### CODE REVIEW

**Comment by Montellese** [comment]
I already suspected that something like this was the problem. Thanks for looking into this. So this bumps us from Android 4.2 to 4.4. Is that acceptable @MartijnKaijser @koying?

@dan1j3l: Please get rid of the merge commit. Could you also add comments to the methods in CJNIViewInputDevice which are not yet available in SDK Version 19 so that they aren't used by accident?

File: xbmc/peripherals/bus/android/PeripheralBusAndroid.cpp [hunk 1]
```diff
@@ -339,13 +339,14 @@ bool CPeripheralBusAndroid::GetDeviceId(const std::string& deviceLocation, int&
...
   int deviceId = inputDevice.getId();
+  int sources = inputDevice.getSources();
   std::string deviceName = inputDevice.getName();
   if (inputDevice.isVirtual())
   {
     CLog::Log(LOGDEBUG, "CPeripheralBusAndroid: ignoring virtual input device \"%s\" with ID %d", deviceName.c_str(), deviceId);
     return false;
   }
-  if (!inputDevice.supportsSource(CJNIViewInputDevice::SOURCE_JOYSTICK) && !inputDevice.supportsSource(CJNIViewInputDevice::SOURCE_GAMEPAD))
+  if (!((sources & inputDevice.SOURCE_GAMEPAD)==inputDevice.SOURCE_GAMEPAD) && !((sources & inputDevice.SOURCE_JOYSTICK)==inputDevice.SOURCE_JOYSTICK))
```
Comment (Montellese) [inline xbmc/peripherals/bus/android/PeripheralBusAndroid.cpp]: Please use `CJNIViewInputDevice::SOURCE_FOO` instead of `inputDevice.SOURCE_FOO`.

Comment (koying) [inline xbmc/peripherals/bus/android/PeripheralBusAndroid.cpp]: and blanks around the "==", please

**Comment by koying** [comment]
+1. Time to bump to 19 now that AFTV is 21

**Comment by MartijnKaijser** [comment]
@dan1j3l thx
Please see https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md  on how to create a pull request as you now have included a merge/master commit. Creating a local branch first solves this problem and makes adding new commits easier without breaking stuff

**Comment by dan1j3l** [comment]
@Montellese Ok, I will make those changes.
@MartijnKaijser Should I just close this PR and make another one (clean one) with all requested changes ?

**Comment by MrMC** [comment]
might whip it through depends at 19 and watch the sparks :)

**Comment by koying** [comment]
@MrMC That's just the Manifest, not the NDK ;)

**Comment by MrMC** [comment]
so, if some lib built under 17, uses a symbol that is not present on user device at 19+, then what happens ? :)

**Comment by koying** [comment]
Nothing ;) 
NDK versions and java api are actually unrelated. The whole raw PT stuff is 21+ and we build it with ndk 17.

File: xbmc/peripherals/bus/android/PeripheralBusAndroid.cpp (hunk 1 above)
Comment (garbear) [inline xbmc/peripherals/bus/android/PeripheralBusAndroid.cpp]: and split into separate lines and De Morganize
```c++
if ((sources & CJNIViewInputDevice::SOURCE_GAMEPAD) != CJNIViewInputDevice:SOURCE_GAMEPAD ||
    (sources & CJNIViewInputDevice::SOURCE_JOYSTICK) != CJNIViewInputDevice:SOURCE_JOYSTICK))
```

**Comment by fritsch** [comment]
Too much bikeshedding or you want to reopen? :-) We look forward to this contribution, so thanks much in advance.

**Comment by dan1j3l** [comment]
@fritsch 
Leave this one closed, later I will create new fork and local clone on my other dev. machine, and do a new PR with code and changes "by the book". :) First time contributing so I need some time to figure everything out. :)

**Comment by fritsch** [comment]
You can contact me on IRC and I can show you how to manage remote branches, do rebasing and so on. Good luck!

**Comment by dan1j3l** [comment]
BTW. After PR is created and I make some other related changes on my local repo. (new commit), should I just push that new commit to origin, or should I rebase that commit and force push to origin rebased commit ?

**Comment by fritsch** [comment]
Origin: Your fork (hopefully), e.g. https://github.com/dan1j3l/xbmc.git
upstream: Us e.g. https://github.com/xbmc/xbmc.git

To add the upstream, do from your checkout:
git remote add upstream https://github.com/xbmc/xbmc.git

Now commit, commit, commit stuff into your local repository (which tracks origin).

After that the great moment comes, when you want to send something upstream, so:
1) fetch upstream: git fetch upstream
2) Rebase your current commits onto us: git rebase -i upstream/master
You can select all the commits you want, writing pick in front of it (default) will add them on top, if you want to squash the commits togehter, e.g. the 2nd and 3rd into the first one, replace the pick with squash
3) when rebasing was succeful, force push to your origin:
git push -f origin nameOfTheBranchYouAreCurrentlyWorking
if rebase failed:
git rebase --abort

A hint: Before rebasing, backup the branch: git branch backupoffeatureBeforeSquashAndRebasing

So, there is no need to close pullrequest, you can just push (-f) in your origin. Rebasing kills history, so this is only allowed for your personal fork.

**Comment by fritsch** [comment]
You can push all the single commits if you want them reviewed, but before merging please squash them together so that the result contains a complete feature and is bisectable, which means: master compiles without your commit or without one of your commits.

**Comment by dan1j3l** [comment]
Ok, thx for extensive explanation ! :) I'll join IRC for any other questions. Now I need to go back to work, so my Kodi "adventure" will resume later when I get home.

**Comment by fritsch** [comment]
No hurry! Thanks that you help us.

**Pull request closed without merging**